### ISSUE TITLE
Ubuntu 20.10 Groovy Proposed: Lightdm-gtk-greeter coredump during boot

### ISSUE THREAD

**Issue by kortewegdevries** [Xubuntu/lightdm-gtk-greeter#77]
This is one of the many core dumps on `journalctl` for a given boot (dumps multiple times before starting)
```
Aug 05 06:30:59 kortewegdevries systemd-coredump[1776]: Process 1719 (lightdm-gtk-gre) of user 114 dumped core.
                                                        
                                                        Stack trace of thread 1747:
                                                        #0  0x00007f8d64ea7c4b __GI_raise (libc.so.6 + 0x40c4b)
                                                        #1  0x00007f8d64e8c855 __GI_abort (libc.so.6 + 0x25855)
                                                        #2  0x00007f8d651dfb2f n/a (libglib-2.0.so.0 + 0x1db2f)
                                                        #3  0x00007f8d6521b399 g_logv (libglib-2.0.so.0 + 0x59399)
                                                        #4  0x00007f8d6521b563 g_log (libglib-2.0.so.0 + 0x59563)
                                                        #5  0x00007f8d65219d89 g_realloc (libglib-2.0.so.0 + 0x57d89)
                                                        #6  0x00007f8d65236e24 n/a (libglib-2.0.so.0 + 0x74e24)
                                                        #7  0x00007f8d65236e87 g_string_sized_new (libglib-2.0.so.0 + 0x74e87)
                                                        #8  0x00007f8d6521b8bb g_log_writer_format_fields (libglib-2.0.so.0 + 0x598bb)
                                                        #9  0x00007f8d6521c896 g_log_writer_standard_streams (libglib-2.0.so.0 + 0x5a896)
                                                        #10 0x00007f8d6521c996 g_log_writer_default (libglib-2.0.so.0 + 0x5a996)
                                                        #11 0x00007f8d6521acf7 g_log_structured_array (libglib-2.0.so.0 + 0x58cf7)
                                                        #12 0x00007f8d6521b139 g_log_default_handler (libglib-2.0.so.0 + 0x59139)
                                                        #13 0x00007f8d6521b37c g_logv (libglib-2.0.so.0 + 0x5937c)
                                                        #14 0x00007f8d6521b563 g_log (libglib-2.0.so.0 + 0x59563)
                                                        #15 0x00007f8d65219ce2 g_malloc (libglib-2.0.so.0 + 0x57ce2)
                                                        #16 0x00007f8d6525c5c6 g_variant_type_new_dict_entry (libglib-2.0.so.0 + 0x9a5c6)
                                                        #17 0x00007f8d6524dc3a g_variant_new_dict_entry (libglib-2.0.so.0 + 0x8bc3a)
                                                        #18 0x00007f8d6545e7d6 n/a (libgio-2.0.so.0 + 0x1137d6)
                                                        #19 0x00007f8d6545efab n/a (libgio-2.0.so.0 + 0x113fab)
                                                        #20 0x00007f8d6545eed7 n/a (libgio-2.0.so.0 + 0x113ed7)
                                                        #21 0x00007f8d6545efab n/a (libgio-2.0.so.0 + 0x113fab)
                                                        #22 0x00007f8d6545f1ec n/a (libgio-2.0.so.0 + 0x1141ec)
                                                        #23 0x00007f8d6545e7bf n/a (libgio-2.0.so.0 + 0x1137bf)
                                                        #24 0x00007f8d6545efab n/a (libgio-2.0.so.0 + 0x113fab)
                                                        #25 0x00007f8d6545eed7 n/a (libgio-2.0.so.0 + 0x113ed7)
                                                        #26 0x00007f8d65461181 g_dbus_message_new_from_blob (libgio-2.0.so.0 + 0x116181)
                                                        #27 0x00007f8d6546bfad n/a (libgio-2.0.so.0 + 0x120fad)
                                                        #28 0x00007f8d653fff29 n/a (libgio-2.0.so.0 + 0xb4f29)
                                                        #29 0x00007f8d653fff6d n/a (libgio-2.0.so.0 + 0xb4f6d)
                                                        #30 0x00007f8d65213e6e g_main_context_dispatch (libglib-2.0.so.0 + 0x51e6e)
                                                        #31 0x00007f8d65214220 n/a (libglib-2.0.so.0 + 0x52220)
                                                        #32 0x00007f8d65214513 g_main_loop_run (libglib-2.0.so.0 + 0x52513)
                                                        #33 0x00007f8d65469e9a n/a (libgio-2.0.so.0 + 0x11ee9a)
                                                        #34 0x00007f8d6523d911 n/a (libglib-2.0.so.0 + 0x7b911)
                                                        #35 0x00007f8d6505a5c9 start_thread (libpthread.so.0 + 0x95c9)
                                                        #36 0x00007f8d64f80693 __clone (libc.so.6 + 0x119693)
                                                        
                                                        Stack trace of thread 1738:
                                                        #0  0x00007f8d65061456 futex_wait_cancelable (libpthread.so.0 + 0x10456)
                                                        #1  0x00007f8d5ae1befb n/a (radeonsi_dri.so + 0x4c5efb)
                                                        #2  0x00007f8d5ae17efb n/a (radeonsi_dri.so + 0x4c1efb)
                                                        #3  0x00007f8d6505a5c9 start_thread (libpthread.so.0 + 0x95c9)
                                                        #4  0x00007f8d64f80693 __clone (libc.so.6 + 0x119693)
                                                        
                                                        Stack trace of thread 1737:
                                                        #0  0x00007f8d65061456 futex_wait_cancelable (libpthread.so.0 + 0x10456)
                                                        #1  0x00007f8d5ae1befb n/a (radeonsi_dri.so + 0x4c5efb)
                                                        #2  0x00007f8d5ae17efb n/a (radeonsi_dri.so + 0x4c1efb)
                                                        #3  0x00007f8d6505a5c9 start_thread (libpthread.so.0 + 0x95c9)
                                                        #4  0x00007f8d64f80693 __clone (libc.so.6 + 0x119693)
                                                        
                                                        Stack trace of thread 1733:
                                                        #0  0x00007f8d65061456 futex_wait_cancelable (libpthread.so.0 + 0x10456)
                                                        #1  0x00007f8d5ae1befb n/a (radeonsi_dri.so + 0x4c5efb)
                                                        #2  0x00007f8d5ae17efb n/a (radeonsi_dri.so + 0x4c1efb)
                                                        #3  0x00007f8d6505a5c9 start_thread (libpthread.so.0 + 0x95c9)
                                                        #4  0x00007f8d64f80693 __clone (libc.so.6 + 0x119693)
                                                        
                                                        Stack trace of thread 1735:
                                                        #0  0x00007f8d65061456 futex_wait_cancelable (libpthread.so.0 + 0x10456)
                                                        #1  0x00007f8d5ae1befb n/a (radeonsi_dri.so + 0x4c5efb)
                                                        #2  0x00007f8d5ae17efb n/a (radeonsi_dri.so + 0x4c1efb)
                                                        #3  0x00007f8d6505a5c9 start_thread (libpthread.so.0 + 0x95c9)
                                                        #4  0x00007f8d64f80693 __clone (libc.so.6 + 0x119693)
                                                        
                                                        Stack trace of thread 1740:
                                                        #0  0x00007f8d65061456 futex_wait_cancelable (libpthread.so.0 + 0x10456)
                                                        #1  0x00007f8d5ae1befb n/a (radeonsi_dri.so + 0x4c5efb)
                                                        #2  0x00007f8d5ae17efb n/a (radeonsi_dri.so + 0x4c1efb)
                                                        #3  0x00007f8d6505a5c9 start_thread (libpthread.so.0 + 0x95c9)
                                                        #4  0x00007f8d64f80693 __clone (libc.so.6 + 0x119693)
                                                        
                                                        Stack trace of thread 1739:
                                                        #0  0x00007f8d65061456 futex_wait_cancelable (libpthread.so.0 + 0x10456)
                                                        #1  0x00007f8d5ae1befb n/a (radeonsi_dri.so + 0x4c5efb)
                                                        #2  0x00007f8d5ae17efb n/a (radeonsi_dri.so + 0x4c1efb)
                                                        #3  0x00007f8d6505a5c9 start_thread (libpthread.so.0 + 0x95c9)
                                                        #4  0x00007f8d64f80693 __clone (libc.so.6 + 0x119693)
                                                        
                                                        Stack trace of thread 1719:
                                                        #0  0x00007f8d64f74d6f __GI___poll (libc.so.6 + 0x10dd6f)
                                                        #1  0x00007f8d6521418e n/a (libglib-2.0.so.0 + 0x5218e)
                                                        #2  0x00007f8d65214513 g_main_loop_run (libglib-2.0.so.0 + 0x52513)
                                                        #3  0x00007f8d6545b4e5 g_dbus_connection_send_message_with_reply_sync (libgio-2.0.so.0 + 0x1104e5)
                                                        #4  0x00007f8d6545b8df n/a (libgio-2.0.so.0 + 0x1108df)
                                                        #5  0x00007f8d6545d8c7 g_dbus_connection_call_sync (libgio-2.0.so.0 + 0x1128c7)
                                                        #6  0x00007f8d660949a7 n/a (liblightdm-gobject-1.so.0 + 0x129a7)
                                                        #7  0x00007f8d66095267 n/a (liblightdm-gobject-1.so.0 + 0x13267)
                                                        #8  0x00007f8d66096be7 n/a (liblightdm-gobject-1.so.0 + 0x14be7)
                                                        #9  0x00007f8d66096e77 n/a (liblightdm-gobject-1.so.0 + 0x14e77)
                                                        #10 0x00007f8d660916e0 n/a (liblightdm-gobject-1.so.0 + 0xf6e0)
                                                        #11 0x00007f8d660919d7 lightdm_user_list_get_users (liblightdm-gobject-1.so.0 + 0xf9d7)
                                                        #12 0x000055f5574fa81e main (lightdm-gtk-greeter + 0xd81e)
                                                        #13 0x00007f8d64e8ea82 __libc_start_main (libc.so.6 + 0x27a82)
                                                        #14 0x000055f5574fb5fa _start (lightdm-gtk-greeter + 0xe5fa)
                                                        
                                                        Stack trace of thread 1731:
                                                        #0  0x00007f8d65061456 futex_wait_cancelable (libpthread.so.0 + 0x10456)
                                                        #1  0x00007f8d5ae1befb n/a (radeonsi_dri.so + 0x4c5efb)
                                                        #2  0x00007f8d5ae17efb n/a (radeonsi_dri.so + 0x4c1efb)
                                                        #3  0x00007f8d6505a5c9 start_thread (libpthread.so.0 + 0x95c9)
                                                        #4  0x00007f8d64f80693 __clone (libc.so.6 + 0x119693)
                                                        
                                                        Stack trace of thread 1736:
                                                        #0  0x00007f8d65061456 futex_wait_cancelable (libpthread.so.0 + 0x10456)
                                                        #1  0x00007f8d5ae1befb n/a (radeonsi_dri.so + 0x4c5efb)
                                                        #2  0x00007f8d5ae17efb n/a (radeonsi_dri.so + 0x4c1efb)
                                                        #3  0x00007f8d6505a5c9 start_thread (libpthread.so.0 + 0x95c9)
                                                        #4  0x00007f8d64f80693 __clone (libc.so.6 + 0x119693)
                                                        
                                                        Stack trace of thread 1734:
                                                        #0  0x00007f8d65061456 futex_wait_cancelable (libpthread.so.0 + 0x10456)
                                                        #1  0x00007f8d5ae1befb n/a (radeonsi_dri.so + 0x4c5efb)
                                                        #2  0x00007f8d5ae17efb n/a (radeonsi_dri.so + 0x4c1efb)
                                                        #3  0x00007f8d6505a5c9 start_thread (libpthread.so.0 + 0x95c9)
                                                        #4  0x00007f8d64f80693 __clone (libc.so.6 + 0x119693)
                                                        
                                                        Stack trace of thread 1732:
                                                        #0  0x00007f8d65061456 futex_wait_cancelable (libpthread.so.0 + 0x10456)
                                                        #1  0x00007f8d5ae1befb n/a (radeonsi_dri.so + 0x4c5efb)
                                                        #2  0x00007f8d5ae17efb n/a (radeonsi_dri.so + 0x4c1efb)
                                                        #3  0x00007f8d6505a5c9 start_thread (libpthread.so.0 + 0x95c9)
                                                        #4  0x00007f8d64f80693 __clone (libc.so.6 + 0x119693)
                                                        
                                                        Stack trace of thread 1730:
                                                        #0  0x00007f8d64f74d6f __GI___poll (libc.so.6 + 0x10dd6f)
                                                        #1  0x00007f8d6521418e n/a (libglib-2.0.so.0 + 0x5218e)
                                                        #2  0x00007f8d652142c3 g_main_context_iteration (libglib-2.0.so.0 + 0x522c3)
                                                        #3  0x00007f8d65214311 n/a (libglib-2.0.so.0 + 0x52311)
                                                        #4  0x00007f8d6523d911 n/a (libglib-2.0.so.0 + 0x7b911)
                                                        #5  0x00007f8d6505a5c9 start_thread (libpthread.so.0 + 0x95c9)
                                                        #6  0x00007f8d64f80693 __clone (libc.so.6 + 0x119693)
Aug 05 06:30:59 kortewegdevries systemd[1]: systemd-coredump@0-1775-0.service: Succeeded.
Aug 05 06:30:59 kortewegdevries kernel: [drm] PCIE GART of 256M enabled (table at 0x000000F400000000).
Aug 05 06:30:59 kortewegdevries kernel: amdgpu: can't get the mac of 5
Aug 05 06:30:59 kortewegdevries audit[1]: SERVICE_STOP pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=systemd-coredump@0-1775-0 comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
Aug 05 06:30:59 kortewegdevries audit: AUDIT1334 prog-id=13 op=UNLOAD
Aug 05 06:30:59 kortewegdevries audit: AUDIT1334 prog-id=12 op=UNLOAD
Aug 05 06:30:59 kortewegdevries audit[1]: SERVICE_STOP pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=lightdm comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=failed'
Aug 05 06:30:59 kortewegdevries systemd[1]: lightdm.service: Main process exited, code=exited, status=1/FAILURE
Aug 05 06:30:59 kortewegdevries systemd[1]: lightdm.service: Failed with result 'exit-code'.
Aug 05 06:30:59 kortewegdevries systemd[1]: lightdm.service: Scheduled restart job, restart counter is at 1.
Aug 05 06:30:59 kortewegdevries systemd[1]: Stopped Light Display Manager.
```

System Details:

```
System: 
  
Kernel: 5.8.0-12-generic x86_64 bits: 64 compiler: N/A Desktop: Xfce 4.14.2 
Distro: Ubuntu 20.10 (Groovy Gorilla) (*Proposed)

lightdm                                                     1.30.0-0ubuntu3.1: amd64        
lightdm-gtk-greeter                                         2.0.6-0ubuntu1:amd64       
lightdm-gtk-greeter-settings                                1.2.2-3       
linux-firmware                                              1.189 
 ---------------------------------------------------
systemd,systemd-coredump,
systemd-sysv                                                245.7-1ubuntu1
---------------------------------------------------                           
OpenGL ES 3.2 
Mesa 20.2.0-devel
```
[lightdm.log](https://termbin.com/eipz)
[coredumpctl](https://termbin.com/chr2)
[gdb_backtrace](https://termbin.com/645a)

Current Behaviour:

On boot screen blinks for 2-3 times (probably the segmentation fault) but eventually starts after 1-2 minutes.

Expected Behaviour:

Greeter would appear properly after boot.

Launchpad (this is where apport took me,I cannot manually file it?): https://bugs.launchpad.net/ubuntu/+source/lightdm-gtk-greeter/+bug/1890394

**Comment by kortewegdevries** [comment]
Moving on to, 
```
# cat /proc/version_signature

Ubuntu 5.8.0-16.17-generic 5.8.0
```
The bug is still there and greeter dumps core almost 2-3 times in a boot before stabilizing.

**Comment by szimszon** [comment]
For me lightdm is not able to start at all. Keeps restarting all day. ;(

**Comment by seb128** [comment]
The issue is fixed with commit dcde7fe8 and in 2.0.7, the Ubuntu package hasn't been updated though, a cherry pick of the fix has been uploaded now

**Comment by kortewegdevries** [comment]
Fixed in Bionic lightdm-gtk-greeter (2.0.5-0ubuntu2), Groovy: 2.0.6-0ubuntu1
